addons/account/models/account_move_line.py

```diff
@@ -1032,6 +1001,15 @@ def _inverse_debit_credit(self):
         for line in self:
             line.balance = line.debit - line.credit
 
+    def _inverse_analytic_distribution(self):
+        """ Unlink and recreate analytic_lines when modifying the distribution."""
+        lines_to_modify = self.env['account.move.line']
+        for line in self:
+            if line.parent_state == "posted":
```

Comment:
This was done for the edit (and multi-edit) of journal items. If I do it for unposted lines then it would create analytic lines if you save an invoice even if you don't post it, if I'm not mistaken